### DISCUSSION TITLE
JPEGVisiualRepairTool save back te same broken image

### DISCUSSION THREAD

**Discussion by Chris-v-d-West** [albmac/JPEGVisualRepairTool#13] [General]
![DSC_7687](https://github.com/albmac/JPEGVisualRepairTool/assets/48785599/f06f81d1-5c98-4d0a-8abb-79b59d0cd905)
I have this image that normally only displays only a small strip at the top.  When I load it in JPEGVisualRepairtool, it automatically displays the full image correctly.  However when I save it, it is identical to the original broken image I am using Chrome in Windows 10, I also tried MicroSoft Edge.

**Comment by albmac** [comment]
Your image has more than one scan frame; you can see by clicking on image info:
![immagine](https://github.com/albmac/JPEGVisualRepairTool/assets/12299278/f26fcd43-05f1-4b23-904b-b96f5a83b5a7)
![immagine](https://github.com/albmac/JPEGVisualRepairTool/assets/12299278/ca877001-46a2-4745-b375-f4b1882568ce)
The current decode function looks at the last SOF0 marker (@0x8CB088) and loads a 1632x1080 picture that is described there.
The first SOF0 @ 0x11A3F is instead where the bigger version starts.
I'm not sure if SOF1 is real or produced by data corruption.
It is a bit unusual to have two frames, but I think it's allowed by the standard.
Anyways, since the current code does not support choosing frames, you can transform the image by deleting the unwanted frame with an hex editor.
If you want the big, damaged one delete from 0x8CB088 to 0x997977; if you want the small one from 0x11888 to 0x8CB088.

**Comment by Chris-v-d-West** [comment]
Thank you.  It is getting a bit complex for me!   This is from a Nikon D3200.  I have looked at other healthy images from the same session and I am also seeing twice SOF0.

**Comment by albmac** [comment]
It is probably a way to have an intermediate resolution version for other purposes.
What about SOF1, do they have this marker as well?
Anyways, here the low resolution image:
![github1](https://github.com/albmac/JPEGVisualRepairTool/assets/12299278/bd3fb127-abe8-4e5b-aced-92d15eed9a28)
And the high resolution one after a bit of alignment and color fix:
![github2_c](https://github.com/albmac/JPEGVisualRepairTool/assets/12299278/fbe3cbeb-7382-4fb1-a0fd-b852b6d9005d)

**Comment by JoepDiskTuna** [comment]
Yes, to restore high resolution image you need some what I'd call technical editing such as done with the jpeg visual repair tool and some photo editing .. 
![fix](https://github.com/albmac/JPEGVisualRepairTool/assets/23461674/6f0d7f82-f8cd-499e-8f11-f40d8b26ff6d)
